src/App.js

```diff
@@ -11,51 +11,124 @@ const generateSquares = () => {
 
   let currentId = 0;
 
-  for (let row = 0; row < 3; row += 1) {
+  for (let row = 0; row < 3; row++) {
     squares.push([]);
-    for (let col = 0; col < 3; col += 1) {
+    for (let col = 0; col < 3; col++) {
       squares[row].push({
         id: currentId,
         value: '',
       });
-      currentId += 1;
-    }
-  }
+      currentId++;
+    };
+  };
 
   return squares;
-}
+};
 
 const App = () => {
 
   const [squares, setSquares] = useState(generateSquares());
+  const [mark, setMark] = useState(PLAYER_1);
+  const [winner, setWinner] = useState('...');
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const togglePlayer = () => {
+    (mark === PLAYER_1) ? setMark(PLAYER_2) : setMark(PLAYER_1);
+  }; // helper function that toggles 'X' or 'O'
 
+  const setSquareValue = (id) => {
+    if (id >= 0 && id <= 2) {
+      squares[0][id].value = mark; // if square is in row 1
```

Comment:
For react-specific reasons, it's actually best not to update state values directly like this.
The alternative approach is to create a new variable and call setSquares using that variable.